src/main/java/com/thebois/models/beings/roles/GuardRole.java

```diff
@@ -10,4 +14,9 @@ public RoleType getType() {
         return RoleType.GUARD;
     }
 
+    @Override
+    protected Collection<IActionGenerator> getTaskGenerators() {
+        return RoleFactory.idle().getTaskGenerators();
```

Comment:
once again i lost, could you breakdown this line?

Reply:
Sure, so `GuardRole` does not have any unique functionality implemented yet. So the behaviour we want is for it to act like an idle being, just randomly moving around the world.

The `getTaskGenerators` method is how different roles provide different sets of action-generators, and these generators then return the actions themselves. In `AbstractRole`, inside `obtainNextAction`, `getTaskGenerators` (the template method) is called when trying to obtain the next action to perform. For `GuardRole`, it comes to this method on line 19.

Here, a new idle role is created from the `RoleFactory`, and then the responsibility of generating `Action`s is delegated to that role. So, in effect, making `GuardRole` behave as if it was an `IdleRole`.

Reply:
👍